docs/modules/ROOT/pages/api/upgrades.adoc

```diff
@@ -75,6 +107,25 @@ Requirements:
 
 Emits an {Upgraded} event.
 
+[.contract-item]
+[[UpgradeableComponent-upgrade_and_call]]
+==== `[.contract-item-name]#++upgrade_and_call++#++(ref self: ContractState, new_class_hash: ClassHash, selector: felt252, calldata: Span<felt252>)++` [.item-kind]#internal#
+
+Replaces the contract's class hash with `new_class_hash` and then calls `selector`
+from the upgraded context.
+
+Requirements:
+
+- `new_class_hash` must be different from zero.
+
+NOTE: The function call comes from the upgraded contract itself and not the account.
+
+NOTE: A similar behavior to `upgrade_and_call` can also be achieved with a list of calls from an account since the {snip6} account standard supports multicall.
+An account can execute a list of calls with xref:IUpgradeable-upgrade[upgrade] being the first element in the list and the extra function call as the second.
+With this approach, the calls will execute from the account's context and can't be front-ran.
```

Comment:
I think we should highlight the difference of these two approaches. With `upgrade_and_call` the caller (returned by `get_caller_address` in the called contract) will be the upgraded contract itself. But with multicall approach it's the account that will be considered to be caller

It's an important difference, because in some cases only one of the approaches will work due to access permissions

Reply:
I included this line in the main description:
"Be aware that the upgraded contract is the caller of `selector` (and not the account)."

Line 125 (highlighted above) expresses that the multicall approach comes from the account. WDYT? I'm open to suggestions if you think we need more

Reply:
I agree it's explained well enough now 👍